src/prpy/planning/vectorfield.py

```diff
@@ -197,32 +217,165 @@ def TerminateMove():
+        @param openravepy.Trajectory traj:  Workspace trajectory,
+                                            represented as an
+                                            OpenRAVE AffineTrajectory.
+        @param float timelimit: Max planning time (seconds).
+        @param float position_tolerance: Constraint tolerance (meters).
+        @param float angular_tolerance:  Constraint tolerance (radians).
+        @param float goal_pose_tol: Goal tolerance (meters).
+
+        @return openravepy.Trajectory qtraj: Configuration space path.
+        """
+        if position_tolerance < 0.0:
+            raise ValueError('Position tolerance must be non-negative.')
+        elif angular_tolerance < 0.0:
+            raise ValueError('Angular tolerance must be non-negative.')
+
+        manip = robot.GetActiveManipulator()
+        Tstart = manip.GetEndEffectorTransform()
+
+        # Get the final end-effector pose
+        duration = traj.GetDuration()
+        T_ee_goal = openravepy.matrixFromPose(traj.Sample(duration)[0:7])
+
+        def vf_path():
+            """
+            Function defining a joint-space vector field.
+            """
+            T_ee_actual = manip.GetEndEffectorTransform()
+
+            # Find where we are on the goal trajectory by finding
+            # the the closest point
+            (_,t) = util.GetMinDistanceBetweenTransformAndWorkspaceTraj(
+                                                                   T_ee_actual,
+                                                                   traj)
+            # Get the desired end-effector transform from
+            # the goal trajectory
+            t_step = 0.2
+            desired_T_ee = openravepy.matrixFromPose(traj.Sample(t)[0:7])
+            pose_ee_next = traj.Sample(t+t_step)[0:7]
+            desired_T_ee_next = openravepy.matrixFromPose(pose_ee_next)
+
+            # The twist between the actual end-effector position and
+            # where it should be on the goal trajectory
+            twist_perpendicular = util.GeodesicTwist(T_ee_actual,
+                                                     desired_T_ee)
+            direction_perpendicular = desired_T_ee[0:3,3] - T_ee_actual[0:3,3]
+
+            # The twist tangent to where the end-effector should be
+            # on the goal trajectory
+            twist_parallel = util.GeodesicTwist(desired_T_ee,
+                                                desired_T_ee_next)
+            direction_parallel = desired_T_ee_next[0:3,3] - desired_T_ee[0:3,3]
+
+            twist = 0.0*twist_perpendicular + 1.0*twist_parallel
+
+            K_p = 0.5
+            direction = (1.0-K_p)*direction_perpendicular + \
+                         K_p*direction_parallel
+            # Normalize the direction vector
+            direction = numpy.array(direction, dtype='float')
+            direction /= numpy.linalg.norm(direction)
+
+            # Modify the twist
+            twist[0:3] = direction
```

Comment:
I'm not sure whats going on here. I think the current implementation will allow arbitrary drift of orientation, since the `(1 - Kp) * direction_parallel` feedback term is only applied to the translational component.